README.md

```diff
@@ -528,9 +518,38 @@ users
 
 > In all the cases, the `user ID` parameter is the unique identifier of the auth0 account instance. i.e. in `google-oauth2|123456789` it would be the part after the '|' pipe: `123456789`.
 
+### Token Validation
+The ID token received as part of the authentication flow is should be verified following the [OpenID Connect specification](https://openid.net/specs/openid-connect-core-1_0.html).
+
+If you are a user of Auth0 Private Cloud with ["Custom Domains"](https://auth0.com/docs/custom-domains) still on the [legacy behavior](https://auth0.com/docs/private-cloud/private-cloud-migrations/migrate-private-cloud-custom-domains#background), you need to override the expected issuer to match your Auth0 domain before starting the authentication.
+
+The validation is done automatically for Web Authentication
+```kotlin
+val account = Auth0("{YOUR_CLIENT_ID}", "{YOUR_CUSTOM_DOMAIN}")
+
+WebAuthProvider.login(account)
+    .withIdTokenVerificationIssuer("https://{YOUR_AUTH0_DOMAIN}/")
+    .start(this, callback)
+```
+
+For Authentication Client, the method `validateClaims()` has to be called to enable it.
+
+```kotlin
+val auth0 = Auth0("YOUR_CLIENT_ID", "YOUR_DOMAIN")
+val client = AuthenticationAPIClient(auth0)
+client
+     .login("{username or email}", "{password}", "{database connection name}")
+     .validateClaims()
```

Comment:
What is the rationale behind adding a method for this instead of having it happen automatically and transparently, while having methods to customize issuer, maxAge, etc?

Reply:
@Widcket that could break users who are still using Auth0 Private Cloud with ["Custom Domains"](https://auth0.com/docs/custom-domains) still on the [legacy behavior](https://auth0.com/docs/private-cloud/private-cloud-migrations/migrate-private-cloud-custom-domains#background)

That is why we have not enabled this by default.

Reply:
But is it worth it compromising the API design over that, instead of pushing the feature to the next major? Because if you make it optional, not many customers will use it. Not even with lots of documentation.

Reply:
This method will be made default and removed in the next major which can handle breaking changes. Until then we wanted to provide an option where the validation can still be made without breaking the current major version.

This is mentioned in AuthenticationRequest.kt Line 54. We have followed a similar behaviour in Java SDKs to provide a feature as optional to make it mandatory in the next release.

Reply:
Thanks for the clarification 👍🏼